cpp/oneapi/dal/backend/primitives/reduction/functors.hpp

```diff
@@ -51,6 +51,38 @@ struct square {
     }
 };
 
+template <typename T>
+struct isinfornan {
+    using tag_t = reduce_unary_op_tag;
+    bool operator()(const T& arg) const {
```

Comment:
Should it be inline?

Reply:
I have no clue unfortunately, was just trying to match the other reduce_unary_op functors like square., suggestions are welcome

Reply:
When I first approached the review of this code I didn't fully understand how those primitives would be used.
Now I see that they are passed as arguments to the reduction primitives.
I think there is no point in inlining this code in this case.